### ISSUE TITLE
Time series: Shared crosshair does not work when tooltip mode set to all

### ISSUE THREAD

**Issue by szerwi** [grafana/grafana#35552]
**What happened**:
Some time ago I have reported that shared crosshair does not work for time series graphs (#33093).
Shared tooltip for time series was added in 8.0.0, but it does not work properly.

When I switch tooltip mode to all in graph's settings,
![image](https://user-images.githubusercontent.com/19311511/121669986-41619800-caad-11eb-85df-97206b7fd062.png)
And then select shared crosshair in dashboard's settings
![image](https://user-images.githubusercontent.com/19311511/121670037-52aaa480-caad-11eb-9758-0219c8fba0d1.png)
It still shares the whole tooltip instead of just the crosshair (the mouse was over the top graph).
![image](https://user-images.githubusercontent.com/19311511/121670183-6fdf7300-caad-11eb-9623-a65e64246d11.png)

**What you expected to happen**:
The other graph should have displayed only the crosshair.

**Environment**:
- Grafana version: 8.0.1
- Data source type & version: Influxdb 2.0.7
- OS Grafana is installed on: Ubuntu
- User OS & Browser: Opera Gx
- Grafana plugins: -
- Others: -

**Comment by UlrichThiess** [comment]
I can confirm that.

**Comment by zuchka** [comment]
confirmed on beta3 and 8.0.1

**Comment by zuchka** [comment]
possibly related to issue https://github.com/grafana/grafana/issues/35266?

**Comment by UlrichThiess** [comment]
I think this means the dashboard setting.

**Comment by zuchka** [comment]
yep!

**Comment by benrubson** [comment]
Still in `8.0.5`.

**Comment by dprokop** [comment]
@benrubson Indeed, this is still not fixed, but we do plan to get that in place soon.

**Comment by nikki-kiga** [comment]
@zuchka can you confirm if this is the expected behavior for the tooltip when the Dashboard is set to "Shared crosshair"?

https://user-images.githubusercontent.com/42276368/127260536-929312cd-720f-49f4-b671-168e15f8cf6b.mov

**Comment by zuchka** [comment]
@nikki-kiga this behavior looks perfect to me! 💯

**Comment by szerwi** [comment]
@nikki-kiga It looks great, but I'm wondering whether there should be points showed on the second graph. It may look strange when the second graph has data that is sent in different time.
Wouldn't it be better to show only vertical and horizontal line on the second graph?

**Comment by nikki-kiga** [comment]
@szerwi good point. Made some changes; the behavior now matches the old graph panel behavior more closely when the Dashboard is set to "Shared crosshair" and Tooltip is set to "All." 

Attaching for reference:

https://user-images.githubusercontent.com/42276368/127515670-ebe3fce1-c354-411e-90ef-458fdad60eb4.mov

**Comment by hexicans** [comment]
Hello,

Always same issue for me : 

1. Old crosshair : 
<img width="927" alt="Capture d’écran 2021-08-29 à 19 19 01" src="https://user-images.githubusercontent.com/13521851/131259369-aea7ea2d-70a3-4964-8f81-b1efcccac9a7.png">


2. New crosshair : 
<img width="554" alt="Capture d’écran 2021-08-29 à 19 19 08" src="https://user-images.githubusercontent.com/13521851/131259374-e221d359-8092-47b3-be16-f587b0838eba.png">


Grafana updated to last version. How to retrieve the old crosshair please ?

Thanks

**Comment by leeoniya** [comment]
@hexicans it's not clear to me what you're trying to show.

you have two examples using two different datasets and different cursor positions. in the top one you're hovering a gap but in the bottom one you are not.

if your complaint is that the top one is showing the nearest datapoint, then please look at https://github.com/grafana/grafana/pull/34552 and the linked issues there and suggest something that will work in both scenarios.

**Comment by szerwi** [comment]
@nikki-kiga I have noticed that shared crosshair option also shares horizontal line.
It looks a bit strange when the 2 graphs have different values in their axes.
Maybe it is worth to change the behavior so only vertical line (which shows time) will be shared?
Current behavior:
1.
![image](https://user-images.githubusercontent.com/19311511/133927993-eec43866-0794-4d72-be2d-48636f439711.png)

2.
![image](https://user-images.githubusercontent.com/19311511/133928068-49d0069d-0625-400a-98d0-ec4f90b2cfb6.png)

And second issue that I found:
Crosshair is not being shared between time series and state timeline panels:
![image](https://user-images.githubusercontent.com/19311511/133928118-ce372d1a-e7b1-4fea-8ac5-72eafc01cd1a.png)
